feat: minimap and full screen map (disabled by default)

gguio opened the pull request:
fixes #64

[LAST GREENFIELD DEMO](https://minimap.mcraft.fun/?mapDir=https%3A%2F%2Fbucket.mcraft.fun%2FGreenfield+v0.5.1%2Fmap-index.json)
[LAST SINGLEPLAYER DEMO](https://minimap.mcraft.fun/?singleplayer=1&version=1.20.4)
- [x] add warp window changes & quick to
- [x] render current player pos on map
- [ ] Render all chunks in single player worlds with all perf optimisations

zardoy commented:
/deploy

zardoy commented:
/deploy

zardoy commented:
/deploy

Commit:
canvases container

Commit:
drawing new canvas after map move

Commit:
drawing new canvas on move

zardoy commented:
/deploy

Commit:
map is drawing on mouse move. Not finished: scaling ruins the position of the map, right formula required

Commit:
full map with overlapping canvases

Commit:
new factor

Commit:
observer component

Commit:
fixed map offset when scaled

Commit:
warp is drawing on right place in fullmap

Commit:
warps are not cut in fullmap

Commit:
canvases clean up

Commit:
add new warp on left click. Fix required: wrong coords on click

Commit:
wrong coords on click fixed

Commit:
new formulas on click. Probelem: if bot is not in the center then click pos is not counted properly

Commit:
fixed click on map. Problem: wrong click coords when scale is applyed

Commit:
Merge branch 'minimap' into fullmap_grid

Commit:
map cell component

Commit:
grid is drawing

Commit:
grid is drawing

Commit:
warp do set on click. Some clean up required but works fine

Commit:
mapcell to mapchunk

Commit:
map offset to point in the middle

Commit:
basic shading

Commit:
smother spinning

Commit:
fix rotation

Commit:
fix misclick for set warp

Commit:
fix offset for warps on minimap

gguio commented:
/deploy

Commit:
dont rotate

Commit:
added async

Commit:
coords fix

Commit:
fix for infinite loop if no blocks

Commit:
resolve conf

Commit:
Merge branch 'minimap_async' into minimap

Commit:
remove gb

zardoy commented:
/deploy

Commit:
full map bigger + minimap rotate

Commit:
async chunk loading for localplayer. Required: move "this" out of async

Commit:
async chunks loading for localplayer. Fix required: map isnt drawing right

Commit:
new grey color for not existing blocks

zardoy committed:
add gzip size logging which is actually more important than regular size for prod build

Commit:
parts of world and warps on a separate canvas

Commit:
warp pos on click fix

Commit:
some minor fixes

zardoy commented:
/deploy

zardoy committed:
add highest block implementation
updated after 'update' event

Commit:
toggle full map in provider

zardoy commented:
/deploy

Commit:
highest block from viewer

Commit:
skip unused map updates

Commit:
show minimap setting

Commit:
setting to enable fullmap and minimap

Commit:
position coords

Commit:
small cleanup

Commit:
app controls in provider

Commit:
double links for discord

Commit:
dropdown discord button in main menu and pause screen

Commit:
no context reset

gguio commented:
/deploy

Commit:
100% map for small screens

Commit:
close button for small screen

Commit:
merge next

Commit:
faster minimap loading

Commit:
fix crash on loading

Commit:
chunk coords

zardoy committed:
fix zooming on desktop Still mobiles doesnt work

Commit:
minimap with scale and fullmap without scale

Commit:
fullmap adjustements

Commit:
set warp scale

zardoy commented:
/deploy

Commit:
merge next

Commit:
fix set warp on full map

Commit:
complete fix for set warp on click

Commit:
known issue: warps dont draw on minimap on server

Commit:
persist warps

Commit:
fix warp info width

Commit:
sync warps for fullmap and minimap on server

Commit:
reset warp on click

Commit:
api to delete warp required

gguio commented:
/deploy

Commit:
resolve

Commit:
fix build

Commit:
fix no idleCallback on mobile

Commit:
fix build

gguio commented:
/deploy

Commit:
remove idleCallback completely

Commit:
fix build

Commit:
right check for requestIdleCallback

gguio commented:
/deploy

Commit:
delete warp functionality

Commit:
some fixes

Commit:
set warp on long touch for mobile

Commit:
Problem: warps are not synced when updated through command line

gguio commented on src/react/MinimapProvider.tsx:
```diff
@@ -15,19 +16,29 @@ export class DrawerAdapterImpl extends TypedEventEmitter<MapUpdates> implements
     super()
     this.playerPosition = pos ?? new Vec3(0, 0, 0)
     this.warps = warps
+    if (localServer) {
```
this subscribe doesnt fire

Commit:
sync server warps with map

Commit:
warp preview on mouse over

Commit:
persist warps on server

Commit:
refactor

gguio commented:
/deploy

Commit:
clean up

Commit:
resolve most warnings

Commit:
move await out of the loop

gguio commented:
/deploy

Commit:
skip updateing pixels when drawing

zardoy commented:
/deploy

Commit:
warp info mobile scale fixed

Commit:
roll back react zoom pan pinch

gguio commented:
/deploy

Commit:
fix fullmap scrolling issue

zardoy committed:
Merge branch 'next' into pr/gguio/147

zardoy committed:
lint

Commit:
loading takes a lot of time + colors are not correct

Commit:
fullmap fix

Commit:
chunks loading queue added

Commit:
hightest block fix

Commit:
viewer fix

Commit:
chunks drawing in localplayer

Commit:
delete old chunks

zardoy committed:
do not duplicate invisible blocks definition 4 times across the codebase!!

zardoy commented:
/deploy

zardoy committed:
add missing file

Commit:
fullmap for localplayer updates

Commit:
clear chunk store

Commit:
chunks loading queue

Commit:
Merge branch 'minimap_useLocalChunks' into minimap

Commit:
fix some canvas didnt render

Commit:
minimap always uses viewer

Commit:
preflat to define block name in localplayer

Commit:
using preflat on server

Commit:
replaceAll fix

zardoy committed:
Merge branch 'next' into minimap

Commit:
unreachable chunks are transparent

Commit:
unreachable chunks are transparent

Commit:
shadows on fullmap

Commit:
resolve

Commit:
global warps in gameAdditionalState

zardoy committed:
Merge branch 'next' into pr/gguio/147

zardoy commented:
/deploy

Commit:
fix colors for newer versions

zardoy commented:
/deploy

Commit:
fix lines in fullmap

gguio commented:
/deploy

Commit:
player pos on different canvas

Commit:
player pos on minimap. Fix required: no player marker on fullmap

Commit:
delete canvases offscreen

Commit:
fix top left chunks loading in fullmap

Commit:
player pos on full map

Commit:
outline for coords in minimap

Commit:
bigger scale for player pos on full map

gguio commented:
/deploy

Commit:
fix default value error in input

Commit:
warps update in fullmap

Commit:
delete updates fullmap. Fixes required: drawing warps causes lagging + after one warp is set then coords on fullmap are broken

Commit:
small fixes

Commit:
clean up

Commit:
fix coords on fullmap are broken

Commit:
interface fixes on set warp window

Commit:
quick tp

gguio committed:
Merge branch 'next' into minimap

Commit:
move bot from fullmap

gguio commented:
/deploy

Commit:
using BlockData for old versions

Commit:
fix low version colors

zardoy commented on src/react/Input.tsx:
```diff
@@ -9,10 +9,10 @@ interface Props extends React.ComponentProps<'input'> {
   const ref = useRef<HTMLInputElement>(null!)
   const [validationStyle, setValidationStyle] = useState<CSSProperties>({})
-  const [value, setValue] = useState(inputProps.value ?? '')
+  const [value, setValue] = useState(defaultValue ?? '')
```
```suggestion
  const [value, setValue] = useState(defaultValue ?? props.value ?? '')
```

zardoy committed:
Update Input.tsx

zardoy commented:
/deploy

Commit:
input fix

zardoy committed:
Merge branch 'next' into minimap

Commit:
small

Commit:
higher tp

Commit:
fix build

Commit:
exclude rsbuildSharedConfig

Commit:
attempt to fix build 1

Commit:
fix quick tp

zardoy commented:
/update

zardoy commented:
/deploy

zardoy commented on src/react/MinimapProvider.tsx:
```diff
@@ -0,0 +1,340 @@
+      this.overwriteWarps(JSON.parse(storageWarps ?? '[]'))
+    }
+    this.isOldVersion = versionToNumber(bot.version) < versionToNumber('1.13')
+    console.log('version is old:', this.isOldVersion)
```
Useless Logging

zardoy committed:
Merge remote-tracking branch 'origin/next' into pr/gguio/147

zardoy committed:
revert mainmenu

zardoy committed:
Fix Setting, [perf] do not update minimap twice and do not update at all if not enabled or opened, fix logging

Commit:
using heightmaps when possible

Commit:
clean up for heightmap

zardoy commented:
/deploy

Commit:
Merge branch 'next' into minimap

Commit:
fixed heightmap name

gguio commented:
/deploy

Commit:
heightmap is better found

gguio commented:
/deploy

Commit:
fix heightmap search

gguio commented:
/deploy

gguio commented:
/deploy

Commit:
fix chunks coords in heightmap

gguio commented:
/deploy

Commit:
heightmap y fix

gguio commented:
/deploy

Commit:
fix open fullmap on mobile

gguio commented:
/deploy

Commit:
Y in position

Commit:
player position on minimap is stable

Commit:
fix color picker for ios

Commit:
small color picker

gguio commented:
/deploy

Commit:
better alignment for color picker

gguio commented:
/deploy

Commit:
faster map redraw

gguio commented:
/deploy

Commit:
Merge branch 'next' into minimap

Commit:
massive refactor of minimapDrawer

Commit:
viewer changes

Commit:
works but map is transparent

Commit:
fix wrong chunks in view

Commit:
chunkStore to map

Commit:
chunks store clear

Commit:
shadows

Commit:
fullmap support

Commit:
different load chunk funcs for fullmap and minimap

Commit:
fullmap loads in singlplayer

Commit:
small clean up

Commit:
mapDrawer moved in addapter

Commit:
fullmap loading in singleplayer

Commit:
loading heightmap from region if possible

Commit:
fix minimap on server

Commit:
fix fullmap on server

Commit:
regular heightmap for fullmap

Commit:
fix build

Commit:
Merge branch 'next' into m_opt

gguio commented:
/deploy

Commit:
fix viewer

Commit:
fix chunk coords in region

Commit:
chunk is loading from heightmap in greenfield

Commit:
fix region

zardoy committed:
Merge branch 'next' into minimap

zardoy commented:
/deploy

zardoy committed:
feat: add a setting to either completely hide perf stats or enable more advanced display
can be used for demos like: `?setting=renderDebug:"none"`

zardoy committed:
fix: fix crafting in singleplayer
fix(generator): leaves were filled with water in new versions
fix(generator): set plains biome for light grass color

zardoy committed:
update types after minecraft-data types update

zardoy committed:
restore a way t ocreate worlds by enter

zardoy committed:
fix: improve playground by allowing sync world for fast iterating of advanced use cases

zardoy committed:
quickly copy your positions with `/pos` in singleplayer

zardoy committed:
autodeploy PRs (#218)

zardoy committed:
fix: correct syntax for checking PR numbers in AUTO_DEPLOY_PRS in preview.yml

zardoy committed:
ci: update variable name from env to vars in AUTO_DEPLOY_PRS check in preview.yml

zardoy committed:
ci: simplify PR number check logic using fromJSON in AUTO_DEPLOY_PRS in preview.yml

zardoy committed:
ci: refactor variable usage from env to vars for PR check in preview.yml

zardoy committed:
ci: enhance trigger conditions for deployment in preview.yml using variables for issue comments and pull requests

zardoy committed:
ci: add push event trigger and refine deployment conditions in preview.yml for improved deployment handling

zardoy committed:
check that!

zardoy committed:
ci: trying to fix auto preview workflow! (#221)

zardoy committed:
ci: update event trigger from pull_request to pull_request_target in preview.yml (fix)

zardoy committed:
ci: refactor deployment workflow in preview.yml by removing unused checks and adding alias retrieval step

zardoy committed:
fix: don’t display advanced stats on prod deploy

zardoy committed:
fix: never get stuck in unloaded chunks! @sa2urami
feat: fully supported spectator mode & basic creative fly fixes

zardoy committed:
feat: 1.21.1 support

zardoy committed:
ci: fix lint

zardoy committed:
should fix publishing to npm

zardoy committed:
fix: don't crash and conflict with g663 spyware installed

zardoy committed:
fix: fix some blocks were rendered as transparent in old versions. Speed up Greenfield renderer by 18% in tunnels.

zardoy committed:
fix: some areas in old world were competely white

zardoy committed:
feat: optimize slabs render performance by rendering less not visible tiles
Improve performance in Greenfield by 6%

zardoy committed:
feat: turn back smooth lighting on old maps

zardoy committed:
feat: server resource packs (#215)

Co-authored-by: Mqx <62719703+Mqxx@users.noreply.github.com>

zardoy committed:
fix: don't make text move in loading screens

zardoy committed:
fix: (latest) or (offline) status text was not displayed sometimes after the page load

zardoy committed:
fix: don't reset the world in mesher after resourece pack textures update

zardoy committed:
feat: format disconnect messages from minecraft servers (display correctly)

fixes #26

zardoy committed:
feat: write published version name on prod website!

zardoy committed:
fix: fix entities display on old versions like 1.8

zardoy committed:
spectator gamemode fixes

zardoy committed:
fix: fix display names in inventories on 1.8

zardoy committed:
fix auth err message

zardoy committed:
send connecting version to auth endpoint

zardoy committed:
do a restoredData validation

zardoy committed:
fix the auth check

zardoy committed:
fox issues with chat on 1.8

zardoy committed:
hotfix: disable fly after going out of spectator mode

zardoy committed:
ci: fix release info writing

zardoy committed:
update npm banner to a working one

zardoy committed:
feat: allow singleplayer mode to be triggered by 'sp=1' query parameter along with `singleplayer=1`

zardoy committed:
docs: add deploy to Koyeb button

zardoy committed:
feat: implement experimental clipWorldBelowY setting for testing

zardoy committed:
fix: Error messages were not displayed after unsuccessful Microsoft auth. Fixed all error messages.

Commit:
last version

Phoenix616 commented:
This looks pretty cool, is there a way to pre-configure whether its enabled or not? (e.g. via a get parameter in the URL) I took a look at the code and didn't find anything in that direction.

zardoy commented:
> This looks pretty cool, is there a way to pre-configure whether its enabled or not? (e.g. via a get parameter in the URL) I took a look at the code and didn't find anything in that direction.

@Phoenix616 Yes, its possible to configure almost **every aspect of the application** via parameters in the URL. Remember that you can change the setting of the application in that way: `?setting=<setting_name>:<setting_value>`. So there is an example link with minimap disabled: https://minimap.mcraft.fun/?mapDir=https%3A%2F%2Fbucket.mcraft.fun%2FGreenfield+v0.5.1%2Fmap-index.json&setting=showMinimap:"never"

@gguio

zardoy committed:
Merge branch 'next' into pr/gguio/147

zardoy committed:
fix minimap crashes

zardoy committed:
not render components when not active to regress the performance

zardoy committed:
fix test

zardoy committed:
fix lint

zardoy commented:
disabled minimap by default because of some issues on some platforms & singleplayer maps.

@Phoenix616 would you be interested to test it? the link is minimap.mcraft.fun (just set minimap setting to always)

zardoy commented:
/deploy

zardoy committed:
restore
